engine/polyglot-api/src/main/scala/org/enso/polyglot/TopScope.scala

```diff
@@ -34,7 +34,24 @@ class TopScope(private val value: Value) {
     value.invokeMember(UNREGISTER_MODULE, qualifiedName): Unit
   }
 
-  def compile(shouldCompileDependencies: Boolean): Unit = {
-    value.invokeMember(COMPILE, shouldCompileDependencies)
+  def compile(
+    shouldCompileDependencies: Boolean
+  ): Unit = {
+    compile(shouldCompileDependencies, None)
   }
+  def compile(
+    shouldCompileDependencies: Boolean,
+    generateDocs: Option[String]
+  ): Unit = {
+    val docsArg = generateDocs.map {
+      case "api" => "api"
+      case _     => "md"
```

Comment:
```suggestion
      case "md"  => "md"
      case other => throw new IllegalStateException("Invalid docs format: " + other)
```

IMO we should fail if the format is invalid, not fall back to a random one.

Otherwise once we add a new format, that can change behaviour of existing workflows, breaking compatibility. This is a more internal option but I still see no point in silently choosing a random option instead of reporting a clear error.